Vector integration in the DblPend example

sarrasoussia opened the pull request:
This PR was originally intended to isolate the dblpend integration work from #4261, but a few small updates to other examples were necessary to ensure the entire project builds successfully.

JacquesCarette commented on code/drasil-example/dblpend/lib/Drasil/DblPend/Body.hs:
```diff
@@ -48,6 +48,9 @@ import Data.Drasil.ExternalLibraries.ODELibraries (scipyODESymbols,
 import Language.Drasil.Code (quantvar)
 import Drasil.DblPend.ODEs (dblPenODEInfo)
 
+-- import Language.Drasil(vec2D, vAdd, vSub, vScale, norm, square)
```
Don't leave commented-out code in files. Just delete it.

JacquesCarette commented on code/drasil-example/dblpend/lib/Drasil/DblPend/DataDefs.hs:
```diff
@@ -122,3 +122,17 @@ forceGQD = mkQuantDef force forceGEqn
+  (sy angularVel_1 $* positionXEqn_1)
+
+velVecEqn_2 :: PExpr
+velVecEqn_2 = vec2D
```
Although this is correct, what we would really want here is a definition at the level of *vectors* rather than its components. We might not be able to 'say' that right now in Drasil, but we should at least 'say' it in a comment.

JacquesCarette commented on code/drasil-example/dblpend/lib/Drasil/DblPend/Expressions.hs:
```diff
@@ -20,6 +19,20 @@ velXExpr_2, velYExpr_2 :: PExpr
+  (sy angularVel_1 $* positionXEqn_1)
+-- Vector2 velocity
+velVecExpr_2 :: PExpr
+velVecExpr_2 = vec2D
```
same here (and likely everywhere in this file): the quantities should be computed 'vectorially' instead of 'component-wise'.

Yes, having everything inside a vector is an improvement, but we would really like to phrase things at the higher level.

sarrasoussia replied:
Could you please clarify what you meant by “vector operations now handled”? Should we be using things like vecAdd, vScale, and deriv explicitly in expressions now? Or were you mainly referring to grouping components into vector quantities like velocityVec_1? Just want to make sure I’m applying it the right way.

JacquesCarette replied:
Yes, exactly (vecAdd, etc).

JacquesCarette reviewed (commented):
As a first step, this is good. Now on to making this not component-wise!

balacij commented:
Hmm, maybe this PR is superseded by #4261?

https://github.com/JacquesCarette/Drasil/pull/4261/files#diff-538bbd015d7596d8b589b6a2553f2f629687b81debba60677e7dbad9a0f88638 (give it a minute to load, but it will bring you to stable/.../DblPend_SRS.html's diff)

B-rando1 committed:
Removed link to deleted wiki page

B-rando1 committed:
Removed link to deleted wiki page

B-rando1 committed:
Added a link to the GOOL/GProc Overview

...to `/wiki/_Sidebar.md`. The placement may be up for discussion, but
under `Folders` seems like a reasonable guess for where to put it.

B-rando1 committed:
Changed link to relative to match other links

B-rando1 committed:
Removed auto-formatting

My formatter switched out the `*`'s for `-`'s, and I didn't notice
until I saw the diff. I'm switching back now.

B-rando1 committed:
Removed auto-formatting

My formatter switched out the `*`'s for `-`'s, and I didn't notice
until I saw the diff. I'm switching back now.

B-rando1 committed:
Replaced `-`'s with `*`'s

As with my other PR, my formatter automatically replaced the latter with
the former, thus making diff comparison impossible. This puts it back,
so the diff should be readable again.

B-rando1 committed:
Replaced `-`'s with `*`'s

As with my other PR, my formatter automatically replaced the latter with
the former, thus making diff comparison impossible. This puts it back,
so the diff should be readable again.

B-rando1 committed:
Removed other random bit of auto-formatting

B-rando1 committed:
Removed other random bit of auto-formatting

JacquesCarette commented on code/drasil-example/dblpend/lib/Drasil/DblPend/Body.hs:
```diff
@@ -70,7 +73,13 @@ mkSRS = [TableOfContents, -- This creates the Table of Contents
        IScope scope,
        IChar [] charsOfReader [],
-       IOrgSec inModel (SRS.inModel [] []) EmptyS],
+  IOrgSec inModel (SRS.inModel [] []) (foldlSent [
```
These new sentences (which are fine) should not be inlined, but defined below and used here.

JacquesCarette commented on code/drasil-example/dblpend/lib/Drasil/DblPend/ODEs.hs:
```diff
@@ -19,35 +20,11 @@ dblPenODEInfo = odeInfo
-          l2  = sy lenRod_2
+    [dbl 1.3463968515384828, exactDbl 0, dbl 2.356194490192345, exactDbl 0] -- unit in radian [3*pi/7, 0, 3*pi/4, 0]
+  [
+    sy angularVel_1,      -- d(theta1)/dt = angular velocity 1
```
Very nice!

JacquesCarette commented on code/drasil-example/pdcontroller/lib/Drasil/PDController/Unitals.hs:
```diff
@@ -2,12 +2,18 @@ module Drasil.PDController.Unitals where
 import Control.Lens ((^.))
 
+-- | Helper function to create Clifford vector spaces of a given dimension
+realVect :: Dimension -> Space
```
This doesn't belong in an example, it should be abstracted out. I'd say in `drasil-lang` for now.

JacquesCarette commented on code/drasil-example/swhs/lib/Drasil/SWHS/Unitals.hs:
```diff
@@ -23,6 +24,10 @@ import Drasil.SWHS.Concepts (water, phsChgMtrl)
 import Control.Lens ((^.))
 
+-- | Helper function to create Clifford vector spaces of a given dimension
+realVect :: Dimension -> Space
```
Same here, abstract out (won't repeat again)

JacquesCarette commented on code/stable/glassbr/src/cpp/ReadTable.cpp:
```diff
@@ -11,21 +11,21 @@ using std::ofstream;
 using std::vector;
 
-void read_table(string filename, vector<double> &z_vector, vector<vector<double>> &x_matrix, vector<vector<double>> &y_matrix) {
+void read_table(string filename, vector<double> &z_vect3DSor, vector<vector<double>> &x_matrix, vector<vector<double>> &y_matrix) {
```
Does this name in the generated code really need to change?

JacquesCarette reviewed (changes requested):
A lot of the more meaningful changes are very good.

There are a number of changes that are not needed and should be reverted. It does seem like this is a good set of changes (overall) to build on. Now you need to slowly re-enable that things you turned off!

B-rando1 committed:
Fixed the wiki link for GOOL/GProc Overview

I had it as an absolute link instead of a local link, and this seems to
mess up how it's displayed when clicked. This should hopefully fix it.

B-rando1 committed:
Wiki sidebar: reverted `*` bullets to `-`

B-rando1 committed:
Wiki sidebar: removed `<` `>` from link

sarrasoussia commented on code/stable/sglpend/SRS/HTML/SglPend_SRS.html:
```diff
@@ -418,7 +418,7 @@ <h2>Purpose of Document</h2>
             <h2>Scope of Requirements</h2>
             <p class="paragraph">
-              The scope of the requirements includes the analysis of a two-dimensional (<span title="two-dimensional">2D</span>) pendulum motion problem with various initial conditions.
+              The scope of the requirements includes the analysis of a two-dimensional (<span title="two-dimensional">2D</span>) pendulum motion problem with various initial conditions using for geometric representation of physical quantities.
```
Removed

sarrasoussia committed:
Fix CI type checker errors by removing position DataDefs

Position vector DataDefs (p_1, p_2) caused type checking errors because
ddENoRefs with PExpr cannot handle vector construction using vect[...]
and cScale operators. The type checker cannot infer vector types in this
context, even though the same expressions work fine in Expressions.hs.

Changes:
- Moved position vector expressions to Expressions.hs (mvPosExpr_1/2)
- Commented out positionVecDD_1/2 from dataDefs list in DataDefs.hs
- Updated GenDefs.hs to remove references to positionVecDD_1
- Updated velocity derivation sentences to not reference DataDefs

Position vectors are still defined and used in:
- Expressions.hs: mvPosExpr_1, mvPosExpr_2
- Unitals.hs: posVec_1, posVec_2
- GenDefs: velocity/acceleration/force derivations

Type checker now passes with no errors.

sarrasoussia committed:
stabilized

sarrasoussia committed:
website stabilized

JacquesCarette committed:
Merge pull request #4408 from JacquesCarette/fix-gool-overview-link

Fixed the wiki link for GOOL/GProc Overview

B-rando1 committed:
Removed `.md` from GOOL/GProc wiki link

B-rando1 committed:
Removed `.md` from GOOL/GProc wiki link

JacquesCarette committed:
Merge pull request #4410 from JacquesCarette/fix-gool-link-2

Removed `.md` from GOOL/GProc wiki link

sarrasoussia committed:
defined magnitude

sarrasoussia committed:
uncomment position vectors to show in datadefs section of the SRS

sarrasoussia committed:
stable

sarrasoussia committed:
used  for scalars multiplication

sarrasoussia committed:
switched from SimpleQDef to ModelQDef for the position vector datadef

sarrasoussia committed:
stable

sarrasoussia committed:
optimized the code for creating vectors

B-rando1 committed:
Updated Stack install guide link

B-rando1 committed:
Updated Stack install guide link

B-rando1 committed:
Removed redundant LaTeX guide link

B-rando1 committed:
Removed redundant LaTeX guide link

B-rando1 committed:
Made the Inkscape links point to latest stable

With this change, 3 of the 4 issues mentioned in #4415 are addressed.

B-rando1 committed:
Made the Inkscape links point to latest stable

With this change, 3 of the 4 issues mentioned in #4415 are addressed.

B-rando1 committed:
Workspace Setup: Removed Graphviz extension link

...for VS Code. It no longer exists, and isn't needed anyway.

B-rando1 committed:
Workspace Setup: Removed Graphviz extension link

...for VS Code. It no longer exists, and isn't needed anyway.

balacij committed:
stabilize projectile SRS artifacts

JacquesCarette committed:
Merge pull request #4416 from JacquesCarette/workspace-setup-links

Fixed Workspace setup broken links

balacij committed:
Move drasil-code's related to searching for an ICO solution scheme to a file separate from CodeSpec.hs

balacij committed:
Move drasil-code's related to searching for an ICO solution scheme to a file separate from CodeSpec.hs

balacij committed:
Remove unused ConfigParser in drasil-code.

Closes #2469

The `ConfigParser.hs` file is both stale, largely incomplete, and tied
to existing `Choices`, which we are going to re-work. If/when
we de-embed data from Drasil, this file would need to be re-designed
anyways, and right now, keeping it in the `drasil-code` folder means we
need to dance around it in `package.yaml`. So, I think we can remove it
for now in favour of making `drasil-code`'s `package.yaml` file a bit
nicer.

balacij committed:
Remove unused ConfigParser in drasil-code.

Closes #2469

The `ConfigParser.hs` file is both stale, largely incomplete, and tied
to existing `Choices`, which we are going to re-work. If/when
we de-embed data from Drasil, this file would need to be re-designed
anyways, and right now, keeping it in the `drasil-code` folder means we
need to dance around it in `package.yaml`. So, I think we can remove it
for now in favour of making `drasil-code`'s `package.yaml` file a bit
nicer.

JacquesCarette committed:
Merge pull request #4418 from JacquesCarette/mvICOSolSrch

Move drasil-code's related to searching for an ICO solution scheme to a file separate from CodeSpec.hs

balacij committed:
Merge branch 'main' of github.com:JacquesCarette/Drasil into rmUnusedCfgPrsr

balacij committed:
Remove Projectile's output message IM

balacij committed:
Remove Projectile's output message IM

B-rando1 committed:
Created wiki page for cold issues

B-rando1 committed:
Created wiki page for cold issues

B-rando1 committed:
Wiki homepage: replaced cold issues text with link

B-rando1 committed:
Wiki homepage: replaced cold issues text with link

balacij committed:
`piSys`: Work with `ChunkDB`s directly rather than through a `System`.

A side-effect of this is that `drasil-system` is no longer a dependency
of `drasil-printers`.

I also cleaned up the exports/imports lists associated with the
`PrintingInformation` data type, and ensured that `PI` (the sole
constructor of `P-I-`) is not exposed.

balacij committed:
`piSys`: Work with `ChunkDB`s directly rather than through a `System`.

A side-effect of this is that `drasil-system` is no longer a dependency
of `drasil-printers`.

I also cleaned up the exports/imports lists associated with the
`PrintingInformation` data type, and ensured that `PI` (the sole
constructor of `P-I-`) is not exposed.

JacquesCarette committed:
Merge pull request #4419 from JacquesCarette/rmUnusedCfgPrsr

Remove unused ConfigParser in drasil-code.

JacquesCarette committed:
Merge pull request #4420 from JacquesCarette/rmProjMsgIm

Remove Projectile's output message IM

JacquesCarette committed:
Merge pull request #4421 from JacquesCarette/wiki-cold-issues

Wiki: Add Cold Issues Page

balacij committed:
Reuse `fullSI` in all examples

`fullSI` is calculated once in all of our `Body.hs` files and then again
in `DocumentLanguage.hs`' `mkDoc` function.

balacij committed:
Reuse `fullSI` in all examples

`fullSI` is calculated once in all of our `Body.hs` files and then again
in `DocumentLanguage.hs`' `mkDoc` function.

JacquesCarette committed:
Merge pull request #4423 from JacquesCarette/piSys

`piSys`: Work with `ChunkDB`s directly rather than through a `System`.

JacquesCarette committed:
Merge pull request #4424 from JacquesCarette/reuseFullSI

Reuse `fullSI` in all examples

JacquesCarette committed:
fix two things noticed by Sam

JacquesCarette committed:
hand-merge main, again again.

JacquesCarette committed:
first step: don't duplicate information from DblPend

JacquesCarette committed:
get rid of some duplication

JacquesCarette committed:
merge re-added this while it was not supposed to be. I prefer to be explicit, but this is not the PR to do this in.

JacquesCarette committed:
Merge pull request #4405 from JacquesCarette/nounPhraseSent4

Noun phrase now does not depend on Sentence at all

balacij committed:
Install fresh JupyterLab copy to fix broken MathJaX links in
JupyterLab-based SRSs converted into HTML

Closes #4339

balacij committed:
Move ExprC CodeExpr typeclass instance to Drasil.Code namespace in
`drasil-lang`

Contributes to #2885

balacij committed:
Remove one (ab)use of mkDocDesc by minimizing its hack.

JacquesCarette committed:
mkDocDesc is now called only once during processing instead of 4 times. This means the extra nub is no longer needed. Part 1 of removing premature automation.

sarrasoussia committed:
added debugging and fixed type checking error; switched back to SimpleQDef in dblpend datadefs

sarrasoussia committed:
Merge branch 'main' into vectorInteg

sarrasoussia committed:
fix hlint errors

balacij committed:
*.hs files: Remove all per-line trailing whitespace.

Done through:

```
find . -type f -name "*.hs" -exec sed -i '' 's/[[:space:]]*$//g' {} +
```

balacij committed:
*.hs files: Ensure all end with a newline character.

JacquesCarette committed:
Merge pull request #4430 from JacquesCarette/mvCodeExprExprC

Move `ExprC CodeExpr` typeclass instance to `Drasil.Code` namespace in `drasil-lang`

JacquesCarette committed:
Merge pull request #4431 from JacquesCarette/fixJupHtml

Install fresh JupyterLab copy to fix broken MathJax JupyterLab-based SRSs converted into HTML

balacij committed:
Merge branch 'main' of github.com:JacquesCarette/Drasil into sglpendWarn

JacquesCarette committed:
Merge pull request #4435 from JacquesCarette/rmTrlngWs

Reduce whitespace-related PR clutter (for future PRs)

balacij committed:
Appease HLint warnings.

balacij committed:
Merge branch 'sglpendWarn' of github.com:JacquesCarette/Drasil into mnmzDcDsc

balacij committed:
Improve comments on traceGCon' and rename to genTraceGraphLabCons.

JacquesCarette committed:
Merge pull request #4433 from JacquesCarette/sglpendWarn

removes extra calls to mkDocDesc

JacquesCarette committed:
Merge pull request #4432 from JacquesCarette/mnmzDcDsc

Remove one (ab)use of mkDocDesc by minimizing its hack.

dependabot[bot] committed:
Bump JamesIves/github-pages-deploy-action from 4.7.3 to 4.7.4

Bumps [JamesIves/github-pages-deploy-action](https://github.com/jamesives/github-pages-deploy-action) from 4.7.3 to 4.7.4.
- [Release notes](https://github.com/jamesives/github-pages-deploy-action/releases)
- [Commits](https://github.com/jamesives/github-pages-deploy-action/compare/v4.7.3...v4.7.4)

---
updated-dependencies:
- dependency-name: JamesIves/github-pages-deploy-action
  dependency-version: 4.7.4
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

B-rando1 committed:
Fixed typos in What-If.md

balacij committed:
Merge pull request #4439 from JacquesCarette/dependabot/github_actions/JamesIves/github-pages-deploy-action-4.7.4

Bump JamesIves/github-pages-deploy-action from 4.7.3 to 4.7.4

B-rando1 committed:
Added 'Working Notes' wiki file

And added it to the sidebar.

B-rando1 committed:
Moved WIP Projects from sidebar to Working Notes

B-rando1 committed:
Moved 'Attributes' discussions

...from Sidebar (under 'Design') to Working Notes

B-rando1 committed:
Working Notes: added headers

Split Other into Design and WIP Projects, to keep the distinction from
the sidebar

B-rando1 committed:
Moved the rest of Working Notes from "Design"

B-rando1 committed:
Added File Input Patterns Discussion

balacij committed:
Re-order cdb arguments to shorten cdb definition

balacij committed:
`Chunk`: Create instance for `HasChunkRefs Chunk`

balacij committed:
`Chunk`: Create instance for `HasChunkRefs Chunk`

balacij committed:
insertAllOutOfOrder12: Only need to check that inserted chunks aren't missing any of their dependencies

balacij committed:
insertAllOutOfOrder12: Add more comments for clarity.

balacij committed:
insertAllOutOfOrder12: Add dependancy checks for LablledContents and References.

balacij committed:
insertAllOutOfOrder12: Fix logical bug in hardLookup. invert will not contain chunks that are not depended on by inserted chunks.

B-rando1 committed:
Added "What If" to homepage

balacij committed:
swhsnopcm: Remove extra added copy of tempW to symbols list.

balacij committed:
Remove `prnt`.

balacij committed:
Re-align `prntDoc` code and note that the lesson plan only renders
Jupyter notebooks and Plain-rendering is generally unsupported.

balacij committed:
Deps clean-up and tidying comment for `srsFormatError`.

balacij committed:
Rename `gen` to `genDoc`.

balacij committed:
Re-align `writeDoc` code.

balacij committed:
`drasil-website`: Don't use `fullSI` when `si` is enough.

balacij committed:
Abstract SRS+code-generation-related code out and move `drasil-gen`.

balacij committed:
insertAllOutOfOrder12: Re-write how chunk dependants lists are updated
for readability.

Now that `insertRefsExpectingExistence` is used, we can remove the
'missing chunk dependancies' check for the main list.

balacij committed:
insertAllOutOfOrder12: Remove previously used function for enumerating all chunks in a database.

balacij committed:
insertAllOutOfOrder12: LCs and Rs barely function as-is in the
`ChunkDB`, so we will remove their dependancy check. Those chunks should
be fixed first.

balacij committed:
drasil-website: Remove duplicate chunks inserted in its chunk database.

balacij committed:
drasil-website: Remove duplicate chunks inserted in its chunk database.

balacij committed:
Switch to out-of-order-permitted chunk insertion when creating `ChunkDB`s

JacquesCarette committed:
Merge pull request #4446 from JacquesCarette/mkCRC

`Chunk`: Create instance for `HasChunkRefs Chunk`

balacij committed:
Remove unused list in SWHS/Body.hs

JacquesCarette committed:
Merge pull request #4444 from JacquesCarette/working-notes

Moved Working Notes pages from Sidebar to new page

JacquesCarette committed:
Merge pull request #4454 from JacquesCarette/dwebch

drasil-website: Remove duplicate chunks inserted in its chunk database.

balacij committed:
`drasil-gen`: Move `docChoices` to Drasil.Generator.Formats and re-wire
exports.

balacij committed:
`drasil-gen`: Move Projectile's `genCodeWithChoices` to `drasil-gen`.

balacij committed:
`drasil-gen`: `genCodeWithChoices` -> `exportSmithEtAlSrsWCodeZoo`

balacij committed:
Remove unused parts of Projectile/Body.hs

balacij committed:
`drasil-gen`: Create `exportLessonPlan` abstracting from Projectile's
Lesson Plan generation code.

balacij committed:
`drasil-gen`: Hide `genDot` export.

It is really an 'internal function' needed for 'complete SRS'
generation, similar to how generating a Makefile is necessary for
generating the mdBook SRS copies.

balacij committed:
`drasil-gen`: Remove `genCode` export.

It is an internal function for `exportCode`.

balacij committed:
`drasil-gen`: Re-organize export lists in `Drasil.Generator.Generate`

balacij committed:
Merge branch 'main' of github.com:JacquesCarette/Drasil into basicTouchUpsOnGen

balacij committed:
`drasil-gen`: abstract away `drasil-website` code related to website
generation.

JacquesCarette committed:
Merge pull request #4449 from JacquesCarette/what-if

Added What If page to Home

JacquesCarette committed:
Merge pull request #4452 from JacquesCarette/swhsnopcmRmTempW

swhsnopcm: Remove extra added copy of tempW to symbols list.

balacij committed:
Merge branch 'main' of github.com:JacquesCarette/Drasil into cdbInsertOutOfOrder12

JacquesCarette committed:
Merge pull request #4445 from JacquesCarette/cdbInsertOutOfOrder12

Switch to out-of-order chunk insertion for the `ChunkDB`

B-rando1 committed:
Updated Windows section of Git2Know

B-rando1 committed:
Added note on VS Code's visual merge editor

B-rando1 committed:
Removed "Things to Avoid" from Git2Know

balacij committed:
Linter: Check for trailing whitespace in Haskell code.

balacij committed:
Clean up missed whitespace cleaning in `drasil-lang`

balacij committed:
Add `check_whitespace` Makefile rule.

balacij committed:
check_whitespace.sh: Avoid using find and a for-loop...

balacij committed:
Fix file permissions for `check_whitespace.sh`.

balacij committed:
`drasil-gen`: Remove `genDoc` export of `drasil-gen`.

JacquesCarette committed:
Merge pull request #4459 from JacquesCarette/git2know

Wiki: Updated Git2know

JacquesCarette committed:
Merge pull request #4464 from JacquesCarette/whitespaceLintCheck

Add trailing whitespace lint step to GitHub Actions workflow and clean remaining trailing whitespace in `drasil-lang`

balacij committed:
`drasil-gen`: Add a few comments on key exported functions.

balacij committed:
Merge branch 'main' of github.com:JacquesCarette/Drasil into basicTouchUpsOnGen

balacij committed:
Appease linter warnings.

balacij committed:
Replace `fillReqs`, `fullReqs` and similar functions that work to
auto-generate FRs with manually-created FRs.

balacij committed:
Remove `inReq EmptyS` hack.

balacij committed:
Replace `FReqsSub` (auto-generating input-values FR) with `FReqsSub'` (no auto-generated input-values FR), removing the latter.

balacij committed:
`ChunkDB`: Enable hard errors on duplicate chunk insertions.

B-rando1 committed:
Added link to New Workspace Setup

B-rando1 committed:
Fixed typo in Workflow.md

B-rando1 committed:
Removed misleading language from PR instructions

B-rando1 committed:
Updated 'Updating Stable' instructions

B-rando1 committed:
Added a clarifying note to PR best practices

B-rando1 committed:
Removed Functions-to-Generate-SRS-Documents.md

JacquesCarette committed:
Merge pull request #4453 from JacquesCarette/basicTouchUpsOnGen

`drasil-gen` touch-ups

JacquesCarette committed:
Merge pull request #4465 from JacquesCarette/protoRmFillReqs

Remove fillReqs, fullReqs and other similar functions that cause "Chunk overwrite" warnings w.r.t. the `ChunkDB`

JacquesCarette committed:
Merge pull request #4467 from JacquesCarette/contributors-guide

Contributor's Guide: Added link to New Workspace Setup

JacquesCarette committed:
Merge pull request #4468 from JacquesCarette/workflow

Wiki: Update Workflow.md

JacquesCarette committed:
Merge pull request #4466 from JacquesCarette/hardErrorsDupeChunkInsert

Hard errors on duplicate chunk insertion

Xinlu-Y committed:
remove deprecated multilingual field from mdBook config

B-rando1 committed:
Fixed/updated some information on Folder Layout

B-rando1 committed:
Removed mention of drasil-code-base

...from Folder-Layout

B-rando1 committed:
Added drasil-system description to Folder Layout

B-rando1 committed:
Added information on stable_website

...to Folder Layout
Contributes to #4461

balacij committed:
`drasil-database`: Remove special treatment of `LabelledContent`.

`LabelledContent` actually have unique `UID`s as they are. The issue
with them, however, is that they inherit them from `Reference`s.
`Reference`s need to be rebuilt so that they don't share the `UID` of
the thing they are 'referring to.'

balacij committed:
`LabelledContent`: Clone the `UID` of the internal `Reference` as its
own.

balacij committed:
GamePhysics: Remove dead code accidentally leftover by #4453.

balacij committed:
`drasil-docLang`: Clean up `mkDoc` a bit.

balacij committed:
`drasil-docLang`: Fuse `fillcdbSRS` with `mkDoc`.

balacij committed:
`exportSmithEtAlSrs`: Do not return `syst'` (`ChunkDB` 'filled' with
references) if `GlassBR` is the only one that needs it
(the whole `Reference` system needs design validation).

balacij committed:
FIXME note about `typeCheckSI`.

balacij committed:
`drasil-docLang`: Explain how `mkDoc` works a bit more.

balacij committed:
`drasil-gen`: `exportSmithEtAlSrs`: No need to return `System` anymore.

balacij committed:
Merge branch 'gpDC' of github.com:JacquesCarette/Drasil into fuseFillCDBMkDoc

B-rando1 committed:
Added information on drasil-codeLang

B-rando1 committed:
Renamed Getting To Know Potential Users of Drasil

(To -> to)

B-rando1 committed:
Added Potential Users to Working Notes

balacij committed:
`drasil-database`/`ChunkDB`: Remove unneeded `UMap`s.

B-rando1 committed:
Added Questions to Working Notes

balacij committed:
Merge branch 'gpDC' of github.com:JacquesCarette/Drasil into mvRefsToSystem

balacij committed:
wip: everything references related to `System` and
`PrintingInformation`. pending some design.

balacij committed:
wip: Remove commented out code from `ChunkDB.hs`.

balacij committed:
wip: Switching backend of `PrintingInformation` to `System` and beg more
code to use `PrintingInformation` rather than `ChunkDB`.

balacij committed:
wip: GlassBR works.

balacij committed:
wip: All other examples compiling and passing stable/ checks.

balacij committed:
Remove unnecessary note.

balacij committed:
Highlight that `drasil-database` now no longer needs any other package
(up to `LabelledContent`/#4481).

balacij committed:
Merge branch 'rmLCSpclTrtmt' of github.com:JacquesCarette/Drasil into drslDBStandAlone

balacij committed:
`ChunkDB.hs`: Get rid of dead code and commentary.

balacij committed:
Move all `drasil-database`-related modules from `drasil-lang` to
`drasil-database`, where they belong!

balacij committed:
`drasil-database`: Move `Dump.hs` to new namespace.

balacij committed:
`drasil-database`: Move `ChunkDB.hs` to new namespace.

balacij committed:
`drasil-database`: Move `TypedUIDRef.hs` to new namespace.

balacij committed:
`drasil-database`: Update re-export module; move to new namespace.

balacij committed:
wip: Fixing `drasil-database`-related imports in other packages.

balacij committed:
`drasil-database`: Fix remaining `drasil-database`-related imports in
all other packages.

balacij committed:
Appease GHC warning about misleading import.

JacquesCarette committed:
Merge pull request #4478 from JacquesCarette/gpDC

GamePhysics: Remove dead code accidentally leftover by #4453.

JacquesCarette committed:
Merge pull request #4480 from JacquesCarette/systPrime

`exportSmithEtAlSrs`: Do not return an altered `System`.

JacquesCarette committed:
Merge branch 'main' into fuseFillCDBMkDoc

JacquesCarette committed:
Merge pull request #4483 from JacquesCarette/folder-layout

Updated Folder layout

dependabot[bot] committed:
Bump actions/checkout from 5 to 6

Bumps [actions/checkout](https://github.com/actions/checkout) from 5 to 6.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

balacij committed:
Merge pull request #4490 from JacquesCarette/dependabot/github_actions/actions/checkout-6

Bump actions/checkout from 5 to 6

B-rando1 committed:
Merge branch 'main' into fuseFillCDBMkDoc

smiths committed:
Merge pull request #4479 from JacquesCarette/fuseFillCDBMkDoc

Fuse `fillcdbSRS` with `mkDoc`.

JacquesCarette committed:
Merge pull request #4485 from JacquesCarette/questions

Added Questions to Working Notes

JacquesCarette committed:
Merge pull request #4487 from JacquesCarette/noMoreUMaps

`drasil-database`/`ChunkDB`: Remove unneeded `UMap`s.

JacquesCarette committed:
Merge pull request #4469 from JacquesCarette/srs-functions

Removed Functions-to-Generate-SRS-Documents.md

JacquesCarette committed:
Merge pull request #4491 from JacquesCarette/mdBookMultiLingual

remove deprecated multilingual field from mdBook config

B-rando1 committed:
Removed "Functions to Generate SRS" From Sidebar

balacij committed:
Merge branch 'main' of github.com:JacquesCarette/Drasil into mvRefsToSystem

JacquesCarette committed:
Merge pull request #4495 from JacquesCarette/srs-functions-sidebar

Removed "Functions to Generate SRS" From Sidebar

balacij committed:
Merge branch 'mvRefsToSystem' of github.com:JacquesCarette/Drasil into drslDBStandAlone

JacquesCarette committed:
Merge pull request #4481 from JacquesCarette/rmLCSpclTrtmt

Remove special treatment of `LabelledContent` in `ChunkDB`.

balacij committed:
Merge branch 'main' of github.com:JacquesCarette/Drasil into mvRefsToSystem

balacij committed:
Merge branch 'mvRefsToSystem' of github.com:JacquesCarette/Drasil into drslDBStandAlone

balacij committed:
`ChunkDB`: Simplify a few functions related to it.

balacij committed:
`ChunkDB`: Update comments.

JacquesCarette committed:
Merge pull request #4488 from JacquesCarette/mvRefsToSystem

Move `Reference` and "reference"-adjacent maps to `System` from `ChunkDB`

JacquesCarette committed:
Merge pull request #4489 from JacquesCarette/drslDBStandAlone

Make `drasil-database` a standalone package (up to `drasil-utils`) and fix module naming scheme

balacij committed:
`drasil-lang`: Switch `RawContent`'s `CodeBlock` constructor from using
`CodeExpr` to `Expr`.

balacij committed:
Remove `drasil-database`-related re-exports from `drasil-lang`.

B-rando1 committed:
Added a line from subpackages to folder layout

B-rando1 committed:
Added a warning message above SubPackages

B-rando1 committed:
Moved link to SubPackages

...from sidebar to working-notes

B-rando1 committed:
Fixed a typo

B-rando1 committed:
Reworked the `git` and `makefile` sections

balacij committed:
Linter Workflow: Add check for excessive consecutive newlines.

balacij committed:
Remove duplicate newline character to trigger Linter CI.

balacij committed:
Remove all extra newlines

Result of executing `find . -type f -name "*.hs" -exec perl -0777 -i -pe 's/\n\n\n/\n\n/g' {} +`.

balacij committed:
`drasil-database`: Update READMEs.

B-rando1 committed:
Updated the notes on debugging Chunks

B-rando1 committed:
Added a note on "not found in"

B-rando1 committed:
Added a note on ripgrep

JacquesCarette committed:
Merge pull request #4497 from JacquesCarette/codeBlockExpr

`drasil-lang`: Switch `RawContent`'s `CodeBlock` constructor from using `CodeExpr` to `Expr`.

JacquesCarette committed:
Merge pull request #4484 from JacquesCarette/potential-users

Updated "Getting to Know Potential Users of Drasil" Page

JacquesCarette committed:
Merge pull request #4499 from JacquesCarette/subpackages

Wiki: Updated SubPackages to Match Its Current Role

JacquesCarette committed:
Merge pull request #4500 from JacquesCarette/lintExcessiveNewlines

Linter Workflow: Add check for excessive consecutive newlines and remove offending newlines.

JacquesCarette committed:
Merge pull request #4505 from JacquesCarette/updbrdm

`drasil-database`: Update READMEs.

JacquesCarette committed:
Merge pull request #4506 from JacquesCarette/wiki-debugging

Wiki: Debugging in Drasil

balacij committed:
Merge branch 'main' of github.com:JacquesCarette/Drasil into drasilDBL

JacquesCarette committed:
Merge pull request #4498 from JacquesCarette/drasilDBL

Remove `drasil-database`-related re-exports from `drasil-lang`.

sarrasoussia committed:
partial fixes

sarrasoussia committed:
WIP: Fix sglpend mass references - use QPP.mass consistently

- Updated GenDefs.hs to use QPP.mass and phrase mass
- Updated Derivations.hs to use QPP.mass in all expressions
- Updated Expressions.hs to use QPP.mass in all expressions
- Cleaned up blank line in Unitals.hs

Remaining issue: Runtime error 'Failed to find chunk mass (expected type: DefinedQuantityDict)'
needs investigation into ChunkDB registration and UID preservation.

sarrasoussia committed:
Merge origin/main into vectorInteg - resolved all conflicts

sarrasoussia committed:
Fix remaining conflict marker in Physics.hs

sarrasoussia committed:
Fix remaining conflict markers in DataDefs.hs and ModuleDefs.hs

sarrasoussia committed:
WIP: Space.hs changes

sarrasoussia committed:
Merge CSchankGA into vectorInteg - resolved all conflicts

JacquesCarette commented:
In theory, I'm supposed to close all "Vector PRs" since they are all unmergeable (including this one). Others are indeed closed.

However this one contains useful work in the changes to `DblPend` that I don't really want to lose. So I'm reluctantly going to leave this one as is for now.

balacij commented:
@JacquesCarette Is there anything specifically that you would want salvaged and incorporated today? Perhaps we can keep this stored as a `.patch` saved within the repo?

JacquesCarette commented:
The useful information is in the files (all under `DblPend`) `Concepts`, `DataDefs`, `Derivations`, `Expressions` and `Unitals`. If that could be kept, then this can be closed.
